lib/cooked_post_processor.rb

```diff
@@ -285,10 +285,11 @@ def create_span_node(klass, content=nil)
     span
   end
 
-  def update_topic_image
-    if @post.is_first_post?
-      img = extract_images_for_topic.first
-      @post.topic.update_column(:image_url, img["src"][0...255]) if img["src"].present?
+  def update_post_image
+    img = extract_images_for_post.first
```

Comment:
The `extract_images_for_post` method filters the image src as per this code: https://github.com/techAPJ/discourse/blob/382803cb052d5251e2a5be2d2b4f18c2c3a72002/lib/cooked_post_processor.rb#L103-L112

That should suffice right?